Fix simple cppcheck issues

matthiakl opened the pull request:
Re #1406
I worked through approx. half of the sources fixing simple cppcheck issues (mostly `const` and `explicit` declarations).
I also updated the `create_cppcheck_report` to work with newest version and github links.

[cppcheck_report.html](https://github.com/widelands/widelands/files/5367268/cppcheck_report.html.txt)

matthiakl committed:
cppcheck in base dir

matthiakl committed:
cppcheck in economy dir

matthiakl committed:
cppcheck in editor dir

matthiakl committed:
cppcheck in graphic dir

matthiakl committed:
cppcheck in io dir

matthiakl committed:
First round of cppcheck in logic dir

matthiakl committed:
cppcheck in logic dir

matthiakl committed:
Merge remote-tracking branch 'upstream/master' into cppcheck

matthiakl committed:
Update cppcheck report

matthiakl committed:
Merge remote-tracking branch 'upstream/master' into cppcheck

matthiakl committed:
Fix clang-tidy issues

actions-user committed:
'src/graphic/text/rt_render.cc' was automatically formatted.

gunchleoc commented:
Testing done, so when the 2 comments are fixed, this can go in.

matthiakl committed:
Small fixes

matthiakl committed:
Merge remote-tracking branch 'upstream/master' into cppcheck

gunchleoc commented:
We have a test failure now as well as a bunch of new compiler warnings https://travis-ci.org/github/widelands/widelands/jobs/735362661

matthiakl committed:
Fix clang warning and testcase

matthiakl commented:
The compiler warning should be fixed, however I can't reproduce it locally (clang-10).
Testcase is also fixed. I added a comment to prevent future cppcheck reviews from the same problem.
+1 for the testsuite beating cppcheck :)

gunchleoc commented:
There's a merge conflict now

matthiakl committed:
Merge remote-tracking branch 'upstream/master' into cppcheck
